### PR TITLE
Run E2E test for `tanzu plugin upload-bundle` against authenticated registry

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it

* Run E2E test for `tanzu plugin upload-bundle` against authenticated registry

NOTE:
* Today Tanzu CLI only supports authenticated registries for the `tanzu plugin upload-bundle` command (push operations).
* But any image pull from the registry requires images to be publically available without requiring authentication
* As part of this test, I am using docker basic authentication to spin up an authenticated registry but only testing the `tanzu plugin upload-bundle` command to verify it works.

Reference: https://github.com/distribution/distribution/blob/main/docs/deploying.md#native-basic-auth

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

* Run existing E2E tests

### Release note

```release-note
None
```

### CODE REVIEW

**Review by chandrareddyp — APPROVED** [review]
LGTM

File: test/e2e/framework/framework.go [hunk 1]
```diff
@@ -92,8 +95,8 @@ func NewFramework() *Framework {
 }
 
 func init() {
-	homeDir := GetHomeDir()
-	TestDirPath = filepath.Join(homeDir, TestDir)
+	OriginalHomeDir = GetHomeDir()
+	TestDirPath = filepath.Join(OriginalHomeDir)
```
Comment (marckhouzam) [inline test/e2e/framework/framework.go:99]: Why are we no longer adding `TestDir` in this `Join`?

File: test/e2e/airgapped/airgapped_test.go [hunk 2]
```diff
@@ -26,6 +27,28 @@ var _ = framework.CLICoreDescribe("[Tests:E2E][Feature:Airgapped-Plugin-Download
 			Expect(err).To(BeNil(), "should not get any error while downloading plugin bundle with specific group")
 		})
 
+		// Test case: upload plugin bundle downloaded using vmware-tkg/default:v0.0.1 plugin-group to the airgapped repository with authentication
+		It("upload plugin bundle that was downloaded using vmware-tkg/default:v0.0.1 plugin-group to the airgapped repository with authentication", func() {
+			curHomeDir := framework.GetHomeDir()
+			defer func() {
+				os.Setenv("HOME", curHomeDir)
+			}()
+			os.Setenv("HOME", framework.OriginalHomeDir)
```
Comment (marckhouzam) [inline test/e2e/airgapped/airgapped_test.go:36]: Why do we need this?  For the docker creds?  Can you put a comment explaining please?

**Review by marckhouzam — COMMENTED** [review]
Looks great, but one concern about the HOME dir before I can approve.

File: test/e2e/airgapped/airgapped_test.go (hunk 2 above)
Comment (anujc25) [reply test/e2e/airgapped/airgapped_test.go:39]: Yes. It is for accessing the docker creds. Added comment.

File: test/e2e/framework/framework.go (hunk 1 above)
Comment (anujc25) [reply test/e2e/framework/framework.go]: Thanks for catching this. I forgot to revert this change back after overwriting the `HOME` variable on the specific test.

**Review by marckhouzam — APPROVED** [review]
LGTM. Thanks

**Pull request merged**